### PR TITLE
turn the "getting started" guide into more of a tutorial

### PR DESCRIPTION
**Pull request by fhars**
### Pull Request Overview

This pull request rewrites part of the "Getting Started" document to be more of a step-by-step tutorial and fixes some steps that did not (no longer?) work (while the memory from #1402 is still fresh).

### Testing Strategy

I tested all steps that were possible with just an nRF52840-DK. 
I checked that the new links in the document point to the correct targets.

### TODO or Help Wanted

Somebody should check that all steps work for other boards, too, and that the necessary changes are described correctly.

It would be good if a native speaker could proofread the text.

### Documentation Updated

- [X] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [X] Ran `make formatall`.

### CODE REVIEW

File: doc/Getting_Started.md [hunk 1]
```diff
@@ -9,6 +9,12 @@ developing Tock.
 1. [Rust](http://www.rust-lang.org/)
 2. [rustup](https://rustup.rs/) to install Rust (version >= 1.11.0)
 3. Command line utilities: make
+4. A supported board.
```
Comment (alistair23) [inline doc/Getting_Started.md]: You could also just use QEMU to get started

File: doc/Getting_Started.md [hunk 2]
```diff
@@ -91,76 +97,153 @@ directory.  All boards share a few common targets:
...
 
-    $ make flash
+### Installing `openocd`
+
+`Openocd` works with various programming and debugging adapters. For
+most purposes, available distribution packages are sufficient and it can
+be installed with:
+
+```bash
+(Linux): sudo apt-get install openocd
```
Comment (alistair23) [inline doc/Getting_Started.md]: This applies only to Debian based distros, not all Linux distros.

Comment (fhars) [reply doc/Getting_Started.md]: It has always been the case that "Linux" in the TockOS documentation means "Ubuntu", I haven't changed that. Should I? And if so, how?

Comment (alistair23) [reply doc/Getting_Started.md]: That seems very confusing. I would just write (Linux/Debian) in the bracket so it's clear

**Pull request merged**